e2etest/bgptests/bgp.go

```diff
@@ -1718,6 +1718,290 @@ var _ = ginkgo.Describe("BGP", func() {
 		table.Entry("IPV6", "fc00:f853:0ccd:e799::/116", "fc00:f853:ccd:e800::1/128", ipfamily.IPv6),
 	)
 
+	ginkgo.Context("IPV4 - Validate service allocation in address pools", func() {
+		ginkgo.AfterEach(func() {
+			// Clean previous configuration.
+			err := ConfigUpdater.Clean()
+			framework.ExpectNoError(err)
+		})
+		ginkgo.It("with namespace", func() {
+			resources := metallbconfig.ClusterResources{
+				Pools: []metallbv1beta1.IPAddressPool{
+					{
+						ObjectMeta: metav1.ObjectMeta{Name: "bgp-test-1"},
+						Spec: metallbv1beta1.IPAddressPoolSpec{
+							Addresses: []string{
+								"192.168.5.0/32",
+							},
+							AllocateTo: &metallbv1beta1.ServiceAllocation{Priority: 20, Namespaces: []string{f.Namespace.Name}},
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{Name: "bgp-test-2"},
+						Spec: metallbv1beta1.IPAddressPoolSpec{
+							Addresses: []string{
+								"192.168.10.0/32",
+							},
+							AllocateTo: &metallbv1beta1.ServiceAllocation{Priority: 10, Namespaces: []string{f.Namespace.Name}},
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{Name: "bgp-test-3"},
+						Spec: metallbv1beta1.IPAddressPoolSpec{
+							Addresses: []string{
+								"192.168.20.0/32",
+							},
+						},
+					},
+				},
+				Peers:   metallb.PeersForContainers(FRRContainers, ipfamily.IPv4),
+				BGPAdvs: []metallbv1beta1.BGPAdvertisement{emptyBGPAdvertisement},
+			}
+
+			for _, c := range FRRContainers {
+				err := frrcontainer.PairWithNodes(cs, c, ipfamily.IPv4)
+				framework.ExpectNoError(err)
+			}
+
+			err := ConfigUpdater.Update(resources)
+			framework.ExpectNoError(err)
+
+			for _, c := range FRRContainers {
+				validateFRRPeeredWithAllNodes(cs, c, ipfamily.IPv4)
+			}
+
+			svc1, _ := testservice.CreateWithBackend(cs, f.Namespace.Name, "external-local-lb-1")
+			svc2, _ := testservice.CreateWithBackend(cs, f.Namespace.Name, "external-local-lb-2")
+			svc3, _ := testservice.CreateWithBackend(cs, f.Namespace.Name, "external-local-lb-3")
+			defer func() {
+				testservice.Delete(cs, svc1)
+				testservice.Delete(cs, svc2)
+				testservice.Delete(cs, svc3)
+			}()
+
+			framework.ExpectEqual(f.Namespace.Name, f.Namespace.Name)
+			for _, i := range svc1.Status.LoadBalancer.Ingress {
```

Comment:
can you add a comment about the fact that this is deterministic because createWithBackend always wait for the service to get an ip?